doc/user/layouts/shortcodes/create-source/syntax-details.html

```diff
@@ -21,6 +22,26 @@
 
 ## Details
 
+### Materialized source details
+
+By default, sources in Materialize do not retain data from the external
+components to which they connect. Instead, [materialized
+views](../../overview/api-components/#materialized-views) and their attendant
+[indexes](../../overview/api-components/#indexes) retain and manage data. The
+presence of these indexes is necessary to make the views `SELECT`-able; the
+absence of indexes are why sources are not `SELECT`-able by default.
+
+However, you can choose to "materialize" a source, which creates an index that
+stores _all_ of the data the source receives. This has the benefit of making the
+source `SELECT`-able, but that must be weighed against the cost of retaining all
+of the source's data for the source's lifetime.
+
+Unlike materialized views, which can innately filter data that is no longer
+pertinent, materialized sources do not offer an equivalent mechanism. Because
+this can lead to materialized sources consuming an unbounded amount of memory,
+they're often best suited for testing and exploratory phases of a Materialize
+deployment.
```

Comment:
These statements look accurate for sources that are not upsert.

Materialized upsert sources filter out deleted/outdated data. There is no benefit to not materializing an upsert source until we add functionality to further filter not-pertinent data (e.g. push filters/projects down into the source).